packages/glsp-playwright-core/src/glsp/features/command-palette/command-palette.base.ts

```diff
@@ -0,0 +1,96 @@
+import type { Locator } from '@playwright/test';
+import { waitForFunction } from '~/integration/wait.fixes';
+import type { GLSPLocator } from '~/remote';
+import { Input, Locateable } from '~/remote';
+
+export interface GLSPBaseCommandPaletteOptions {
+    locator: GLSPLocator;
+    suggestionLocator?: GLSPLocator;
+    loadingLocator?: GLSPLocator;
+}
+
+export abstract class GLSPBaseCommandPalette extends Locateable {
+    readonly input;
+    readonly suggestionLocator;
+    readonly loadingLocator;
+
+    protected readonly keyboard;
+
+    constructor(protected readonly options: GLSPBaseCommandPaletteOptions) {
+        super(options.locator);
+
+        this.suggestionLocator = options.suggestionLocator ?? this.locator.child('.command-palette-suggestions');
+        this.loadingLocator = options.loadingLocator ?? this.locator.child('.loading');
+
+        this.keyboard = this.app.page.keyboard;
+
+        this.input = new Input(this.locator.child('input'));
+    }
+
+    abstract open(): Promise<void>;
+
+    async search(text: string, options?: { confirm?: boolean }): Promise<void> {
+        const suggestions = await this.suggestions();
+        await this.input.type(text, { delay: 50 });
+        await this.waitForLoadingDetached();
+        await this.waitForSuggestionsChange(suggestions);
+
+        if (options?.confirm) {
+            await this.input.press('Enter');
+            await this.waitForHidden();
+        }
+    }
+
+    async confirm(suggestion: string): Promise<void> {
+        await this.suggestionLocator.locate().locator(`text=${suggestion}`).click();
+        await this.waitForHidden();
+    }
+
+    async suggestions(): Promise<string[]> {
+        await this.waitForLoadingDetached();
+        await this.waitForSuggestionsChange();
+
+        return this.suggestionLocator
+            .locate()
+            .locator('> *')
+            .evaluateAll(elements =>
+                elements.map(element => {
+                    const content = element.textContent;
```

Comment:
It would be nice if that textual cleanup is in a dedicated protected method so it could be easily extended if necessary.